### DISCUSSION TITLE
Is there anyways to skip non existent record while connecting

### DISCUSSION THREAD

**Discussion by bigint** [prisma/prisma#9462] [Q&A]
Is there anyways to skip non existent record while connecting

![image](https://user-images.githubusercontent.com/69431456/134889171-a87154f5-cf4e-493a-bc8f-696f3a59555f.png)

Out of this one username is correct two is wrong, I need to skip those two usernames while connecting 😕 

```ts
import { getMentions } from '@graphql/utils/getMentions'
import { parseMentions } from '@graphql/utils/parseMentions'
import { db } from '@utils/prisma'

import { Post, Session } from '.prisma/client'

export const processMentions = async (
  post: Post,
  session: Session | undefined | null
) => {
  const users = parseMentions(getMentions(post?.body))

  await db.notification.create({
    data: {
      dispatcher: { connect: { id: session?.userId } },
      receiver: { connect: users },
      type: 'USER_MENTION',
      entityId: post?.id
    }
  })
}
```

**Comment by ryands17** [comment]
@bigint 👋 

You would need to conditionally add the `connect` in this case:

```ts
await db.notification.create({
    data: {
      ...(session?.userId && { dispatcher: { connect: { id: session.userId } } }),
      receiver: { connect: users },
      type: 'USER_MENTION',
      entityId: post?.id
    }
})
```

**Reply by bigint** [reply]
Ahh @ryands17, thanks! But we have noting to do with `dispatcher` 😄 the problem is with `receiver: { connect: users },` where users have array of usernames `[{username: 'ryands17'}, {username: 'bigint'}, {username: 'nonexistantusername'}]` where we need to skip `nonexistantusername` from inserting

Hope you got it!

**Reply by ryands17** [reply]
As far as I know, this isn't possible. An error will always be thrown if a record doesn't exist, so you would need to check for that beforehand.

**Reply by bigint** [reply]
Makes sense @ryands17, thanks!

**Reply by ryands17** [reply]
@bigint 👋 

It would be great if you could create a feature request [here](https://github.com/prisma/prisma/issues/new?assignees=&labels=&template=feature_request.md&title=) if you'd like this to be implemented 🙂

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
